cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java

```diff
@@ -1053,6 +1055,45 @@ public void testSearchMetadata() throws Exception {
     }
   }
 
+  @Test
+  public void testCrossNamespaceSearchMetadata() throws Exception {
+    NamespaceId namespace1 = new NamespaceId("ns1");
+    namespaceClient.create(new NamespaceMeta.Builder().setName(namespace1).build());
+    NamespaceId namespace2 = new NamespaceId("ns2");
+    namespaceClient.create(new NamespaceMeta.Builder().setName(namespace2).build());
+
+    try {
+      appClient.deploy(namespace1, createAppJarFile(AllProgramsApp.class));
+      appClient.deploy(namespace2, createAppJarFile(AllProgramsApp.class));
+
+      // Add metadata to app
+      Map<String, String> props = ImmutableMap.of("key1", "value1");
+      Metadata meta = new Metadata(props, Collections.emptySet());
+      ApplicationId app1Id = namespace1.app(AllProgramsApp.NAME);
+      addProperties(app1Id, props);
+
+      ApplicationId app2Id = namespace2.app(AllProgramsApp.NAME);
+      addProperties(app2Id, props);
+
+      MetadataSearchResponseV2 results = searchMetadata(null, "value*", EnumSet.allOf(EntityTypeSimpleName.class),
+                                                        null, 0, 10, 0, null, false, true);
+
+      Map<MetadataEntity, Metadata> expected = new HashMap<>();
+      expected.put(app1Id.toMetadataEntity(), meta);
+      expected.put(app2Id.toMetadataEntity(), meta);
+
+      Map<MetadataEntity, Metadata> actual = new HashMap<>();
+      Assert.assertEquals(2, results.getResults().size());
+      for (MetadataSearchResultRecordV2 record : results.getResults()) {
```

Comment:
@rohitsinha54 Don't use `forEach` to mutate collection. Use collector instead.

Reply:
For Java stream style, it would be

```
    Map<MetadataEntity, Metadata> actual = results.getResults()
      .stream().collect(Collectors.toMap(MetadataSearchResultRecordV2::getMetadataEntity, 
                                         r -> r.getMetadata().get(MetadataScope.USER)));

```

However, I think for loop is kind of ok in this case.

Reply:
I think the for loop is easier to read, just going to keep it.